### PR TITLE
.github: update s3tests workflow to use new dev env

### PR DESCRIPTION
**Pull request by evgeniiz321**

### CODE REVIEW

File: .github/workflows/s3-tests.yml [hunk 1]
```diff
@@ -84,90 +106,50 @@ jobs:
       - run: go version
 
       - name: Set up Python
-        uses: actions/setup-python@v5
+        uses: actions/setup-python@v4
```
Comment (smallhive) [inline .github/workflows/s3-tests.yml:109]: Is v4 more suitable?

Comment (evgeniiz321) [reply .github/workflows/s3-tests.yml]: new is always better :) it was a copypaste from somewhere

File: .github/workflows/s3-tests.yml [hunk 2]
```diff
@@ -42,38 +42,60 @@ jobs:
...
+          repo: 'nspcc-dev/neofs-node'
+          version: 'tags/v0.41.1'
+          file: 'neofs-node-linux-amd64'
+          target: 's3-tests/neofs-node'
+
+      - name: Download latest stable neo-go
+        uses: dsaltares/fetch-gh-release-asset@1.1.1
+        with:
+          repo: 'nspcc-dev/neo-go'
+          version: 'tags/v0.104.0'
```
Comment (roman-khimov) [inline .github/workflows/s3-tests.yml:96]: v0.105.1 at least.

Comment (evgeniiz321) [reply .github/workflows/s3-tests.yml]: Updated.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**